File: app/src/main/java/org/mifos/mobilebanking/api/services/NotificationService.java
package org.mifos.mobilebanking.api.services;

import org.mifos.mobilebanking.api.ApiEndPoints;
import org.mifos.mobilebanking.models.notification.NotificationRegisterPayload;
import org.mifos.mobilebanking.models.notification.NotificationUserDetail;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

/**
 * Created by dilpreet on 26/09/17.
 */

public interface NotificationService {

    @GET(ApiEndPoints.DEVICE + "/registration/client/{clientId}")
    Observable<NotificationUserDetail> getUserNotificationId(@Path("clientId")long clientId);

    @POST(ApiEndPoints.DEVICE + "/registration")
    Observable<ResponseBody> registerNotification(@Body NotificationRegisterPayload payload);

    @PUT(ApiEndPoints.DEVICE + "/registration/{id}")
    Observable<ResponseBody> updateRegisterNotification(@Path("id")long id,
                                                  @Body NotificationRegisterPayload payload);
}

File: app/src/main/java/org/mifos/mobile/api/services/NotificationService.kt
package org.mifos.mobile.api.services

import io.reactivex.Observable
import okhttp3.ResponseBody
import org.mifos.mobile.api.ApiEndPoints
import org.mifos.mobile.models.notification.NotificationRegisterPayload
import org.mifos.mobile.models.notification.NotificationUserDetail
import retrofit2.http.*

/**
 * Created by dilpreet on 26/09/17.
 */
interface NotificationService {
    @GET(ApiEndPoints.DEVICE + "/registration/client/{clientId}")
    suspend fun getUserNotificationId(@Path("clientId") clientId: Long): NotificationUserDetail

    @POST(ApiEndPoints.DEVICE + "/registration")
    suspend fun registerNotification(@Body payload: NotificationRegisterPayload?): ResponseBody

    @PUT(ApiEndPoints.DEVICE + "/registration/{id}")
    suspend fun updateRegisterNotification(
        @Path("id") id: Long,
        @Body payload: NotificationRegisterPayload?,
    ): ResponseBody
}
